sdk/clientcore/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -135,7 +137,7 @@ private SSLSocketFactory getSslSocketFactory() {
      */
     private HttpURLConnection connect(HttpRequest httpRequest) throws IOException {
         HttpURLConnection connection;
-        URL url = httpRequest.getUrl();
+        URL url = httpRequest.getUri().toURL();
```

Comment:
Need to handle the `MalformedURLException` that the `toURL()` method can throw.

Reply:
Also, doesn't this add extra performance overhead as it converts the URI into a URL?

Reply:
`MalformedURLException` is a subtype of `IOException`, so that should be handled by the checked throws in this method. Previously, we'd never reach this point with a malformed URL, which that's possible now with this change I don't think it be much of a runtime difference.

And yes, this does introduce additional overhead as we need to convert from URI to URL but this is following guidance from the JDK itself: https://docs.oracle.com/en/java/javase/21/docs/api//java.base/java/net/URL.html#constructor-deprecation